src/Shared/Components/DynamicDataTable/DynamicDataTableRow.tsx

```diff
@@ -71,6 +76,7 @@ export const DynamicDataTableRow = <K extends string, CustomStateType = Record<s
     }
 
     useEffect(() => {
+        setIsRowAdded(rows.length && Object.keys(cellRef.current).length < rows.length)
```

Comment:
I think we might end up adding 0
```suggestion
        setIsRowAdded(rows.length > 0 && Object.keys(cellRef.current).length < rows.length)
```